src/AnnotatedCode.h

```diff
@@ -2,15 +2,28 @@
 #ifndef R2GHIDRA_ANNOTATEDCODE_H
 #define R2GHIDRA_ANNOTATEDCODE_H
 
+#include <r_core.h>
 #include <r_types.h>
 #include <r_vector.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef enum r_syntax_highlight_type_t {
+  R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD,
+  R_SYNTAX_HIGHLIGHT_TYPE_COMMENT,
+  R_SYNTAX_HIGHLIGHT_TYPE_DATATYPE,
+  R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME,
+  R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_PARAMETER,
+  R_SYNTAX_HIGHLIGHT_TYPE_LOCAL_VARIABLE,
+  R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE,
+  R_SYNTAX_HIGHLIGHT_TYPE_GLOBAL_VARIABLE,
+} RSyntaxHighlightType;
```

Comment:
I think it might be better to have all these as separate entries in RCodeAnnotationType and without SYNTAX_HIGHLIGHTING in the name. This way we can also add more info, for example for a variable a reference to the actual value which we can use for example for right-click in Cutter.
Or do you think this fits ghidra's xml better?

Reply:
Looking at the XML, I think your current approach is good. But I'd still like to hear your opinion.

Reply:
The current implementation fits 1:1 to ghidra's XML data model. However, I don't think it is suitable for anything other than syntax highlighting.

The main problem is that the granularity is too coarse. For example, both string literals and integer constants are classified as `R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE` by ghidra. So if we want to treat strings and integers in a different manner (e.g. in Cutter with a click or to highlight them in different colours), this will need additional parsing. There are other examples where finer granularity might also be needed (e.g. differentiate between imported vs not-imported symbols).

What you purpose is a good idea. The question is whether we want to stay constrained by ghidra's XML output or we should simply lex the decompiled code and annotate it ourselves. This is more powerful and gives is enough flexibility for future extensions (e.g. we can more easily make the decompiled code to be interactive).

Reply:
For now, I think we can leave the current implementation as it is to provide at least syntax highlighting. Later we can start a separate discussion on how to add a more powerful type annotation for `RAnnotatedCode` (and what we can use it for).

Reply:
Yes agreed. One possibility would also be to have both "actual" semantic annotations and syntax-highlighting annotations in parallel. This can make code that only needs syntax highlighting easier to write and semantic annotations can always easily be converted to syntax highlighting ones if needed.